src/base/misc/fatfs.c

```diff
@@ -501,6 +503,12 @@ static void update_geometry(fatfs_t *f, unsigned char *b)
     // drive number set
     b[0x1fd] = f->drive_num;
   }
+
+  if (version >= 400)
+    memcpy(bpb->v400_fat_type, f->fat_type == FAT_TYPE_FAT12 ? "FAT12   " : "FAT16   ", 8);
```

Comment:
Shouldn't this be also under the MOS_D check?

Reply:
No actually it's a general omission that I spotted for the with boot.blk case. I suppose it should have been a separate commit.